### DISCUSSION TITLE
Gcode example

### DISCUSSION THREAD

**Discussion by maxime-granger** [smartin015/continuousprint#68] [General]
I am just starting with Continuous Print.  
I am using a Prusa Mk3s +

I am no expert and I would like to avoid scratching my bed using a not so good Gcode to clear the bed. 
Is there a template of Gcode for the Bed cleaning script & Queue finished script somewhere I can start with. 

Thank you

**Comment by smartin015** [comment]
Hey, thanks for the post - I don't have an Mk3s+, but it looks like there's a few examples of bed clearing floating around the internet (e.g. [this one](https://www.youtube.com/watch?v=FMwFJGxN24M)).

You can read more on CPQ's stored bed clearing scripts at https://smartin015.github.io/continuousprint/gcode-scripting/. In particular, it'd be great if you could contribute a printer profile with scripts so other Mk3s+ users can just select from a drop-down. 

HTH

**Comment by maxime-granger** [comment]
thanks for this link. I will do my best .

**Comment by rogiepoo** [comment]
Hi! Use at your own risk, but: 
M104 T0 S0 ; extruder heater off
M140 S0 ; heated bed heater off
G28 ; home all axes
G90 ; absolute positioning
G0 X25 Y0 Z20 F3000 ; raise the extruder 2cm go to COOLING location at 3000% normal speed
M190 R27 ; wait for bed to cool to 27C
G28 ; home all axes
G90 ; absolute positioning
G0 X25 Y5 Z20 F3000 ; raise the extruder 2cm go to starting point at 3000% normal speed
G0 X245 Y5 Z20 F3000 ; go to lower right corner of bed, hold Z at 20
G0 X245 Y275 Z20 F3000 ; go to upper right of bed, holding Z at 20
G0 X215 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X215 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X215 Y5 Z1 F1000 ; slowly complete first wipe vector
G0 X215 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X215 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X185 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X185 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X185 Y5 Z1 F1000 ; slowly complete 2nd wipe vector
G0 X185 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X185 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X155 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X155 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X155 Y5 Z1 F1000 ; slowly complete 3rd wipe vector
G0 X155 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X155 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X125 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X125 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X125 Y5 Z1 F1000 ; slowly complete 4th wipe vector
G0 X125 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X125 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X95 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X95 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X95 Y5 Z1 F1000 ; slowly complete 5th wipe vector
G0 X95 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X95 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X65 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X65 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X65 Y5 Z1 F1000 ; slowly complete 6th wipe vector
G0 X65 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X65 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X35 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X35 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X35 Y5 Z1 F1000 ; slowly complete 6th wipe vector
G0 X35 Y5 Z20 F3000: lift Z 20mm up @ 3000%speed
G0 X35 Y275 Z20 F3000 ; go to top of bed, holding Z at 20
G0 X5 Y275 Z20 F3000 ; go over to the left by 3 cm
G0 X5 Y275 Z1 F3000 ; drop down the nozzle to 1mm over the bed
G0 X5 Y5 Z1 F1000 ; slowly complete 6th wipe vector
G28 ; home all axes

I keep this in the Continuous Print "Bed Clearing" script.  But I have an issue and that is my printer enters into mesh bed leveling three or four times before beginning the next print in queue.  However the script above works pretty well in and of itself. It is very important to consider the height of your Z offset as well as Z height within the clearing script. Set at least the z in the clearing script high, like 10mm, and then you can always modify that later.  This one is at 1mm. I have a bed overlay that is accommodated by my high Z offset of 0.880. 
rr
